src/staticwebapp/azext_staticwebapp/_utils.py

```diff
@@ -0,0 +1,313 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+from enum import Enum, auto
+import re
+from functools import lru_cache
+
+from msrestazure.tools import parse_resource_id
+from knack.log import get_logger
+from azure.cli.core.azclierror import InvalidArgumentValueError, RequiredArgumentMissingError, ValidationError
+from azure.cli.core.util import send_raw_request
+from azure.cli.command_modules.cosmosdb._client_factory import cf_db_accounts
+
+
+logger = get_logger(__name__)
+
+
+class ConnectionType(Enum):
+    CONNECTION_STRING = auto()
+    MANAGED_IDENTITY_USER_ASSIGNED = auto()
+    MANAGED_IDENTITY_SYSTEM_ASSIGNED = auto()
+
+
+class Sku(Enum):
+    FREE = auto()
+    STANDARD = auto()
+
+
+class AbstractDbHandler:
+    DB_TYPE_NAME = ""  # used for emitting user-friendly error messages
+    API_VERSION = ""  # the API version for performing GET on RID
+
+    @classmethod
+    def _requires_username(cls, sku: 'Sku', connection_type: 'ConnectionType') -> bool:
+        raise NotImplementedError()
+
+    @classmethod
+    def _requires_password(cls, sku: 'Sku', connection_type: 'ConnectionType') -> bool:
+        raise NotImplementedError()
+
+    @classmethod
+    def _is_supported(cls, sku: 'Sku', connection_type: 'ConnectionType') -> bool:
+        raise NotImplementedError()
+
+    @classmethod
+    def _requires_database_name(cls) -> bool:
+        raise NotImplementedError()
+
+    @classmethod
+    def _get_connection_string(cls, cmd, sku: 'Sku', connection_type: 'ConnectionType', resource_id, database_name,
+                               username=None, password=None, **kwargs):
+        raise NotImplementedError()
+
+    # saves some time by prevent reparsing of RIDs
+    @classmethod
+    @lru_cache(maxsize=100)
+    def _parse_resource_id(cls, resource_id: str) -> dict:
+        return parse_resource_id(resource_id)
+
+    @classmethod
+    def _validate(cls, sku: 'Sku', connection_type: 'ConnectionType', database_name, username, password):
+        if not cls._is_supported(sku, connection_type):
+            raise ValidationError(f"Authentication type '{connection_type}' is not supported for "
```

Comment:
It is better to point out what types are supported.

Reply:
Thanks for the suggestion, just added that